### PR TITLE
[SD-4093] - Search results didn't display the last corrected item

### PR DESCRIPTION
**Pull request by akintolga**

### CODE REVIEW

File: scripts/superdesk-search/search.js [hunk 1]
```diff
@@ -2050,13 +2070,14 @@
                             var itemsById = angular.extend({}, listComponent.state.itemsById);
 
                             items._items.forEach(function(item) {
-                                if (!itemsById[item._id] || !isSameVersion(itemsById[item._id], item)) {
-                                    itemsById[item._id] = item;
+                                var itemId = search.generateTrackByIdentifier(item);
```
Comment (mdhaman) [inline scripts/superdesk-search/search.js]: We get `_id` of the document in the item:lock and item:unlock notification. 
https://github.com/akintolga/superdesk-client-core/blob/cd-4093/scripts/superdesk-search/search.js#L2098
we won't be able to find the item in the `itemsById[itemId]` to update the state.

Comment (akintolga) [inline scripts/superdesk-search/search.js]: That's correct. Now when we lock we use the `id:version` and when an unlock happens we use `updateAllItems` function to update all previous versions to unlock. This is required in published collection where different versions of the same story is visible and when an update happens, returned version is different than the locked one.

**Comment by akintolga** [comment]
depends on https://github.com/superdesk/superdesk-core/pull/210

**Comment by petrjasek** [comment]
either this one or https://github.com/superdesk/superdesk-client-core/pull/91 will need an update, lets see which one gets merged first ;)

**Comment by petrjasek** [comment]
I have a question in general - is there a usecase when there would be multiple versions of same item visible at once?

**Comment by akintolga** [comment]
Multiple versions of the same item are visible when the search is done against the "published" collection. They have the same id (backend does the flicking...)

**Comment by mdhaman** [comment]
:+1:

**Comment by marwoodandrew** [comment]
:+1:

**Comment by akintolga** [comment]
Hey Petr, I'll leave "which one goes in first decision" to you... I'm pretty sure you'll do the right thing :)

**Comment by petrjasek** [comment]
well seems like @ioanpocol decided ;) I will update my version

**Comment by ioanpocol** [comment]
@petrjasek  In the morning your PR already needed a rebase so I merged the other one.

**Comment by petrjasek** [comment]
@ioanpocol sure no problem

**Pull request merged**